src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -187,12 +187,15 @@ private void processRequests(HttpServerRequest httpServerRequest) {
                 final SinkBridgeEndpoint deleteSinkEndpoint = this.httpSinkEndpoints.get(deleteInstanceID);
 
                 if (deleteSinkEndpoint != null) {
-
                     deleteSinkEndpoint.handle(new HttpEndpoint(httpServerRequest));
 
                     this.httpSinkEndpoints.remove(deleteInstanceID);
                 } else {
-                    throw new NullPointerException("no conusmer instance found with this id");
+                    httpServerRequest.response()
+                            .setStatusCode(404)
+                            .putHeader("Content-length", String.valueOf("".length()))
+                            .write("")
```

Comment:
when the body is empty isn't enough calling `end()` avoiding an header with "Content-length" 0 and writing an empty string? Just asking, have you already tried?